packages/react-native-windows-init/src/Cli.ts

```diff
@@ -0,0 +1,291 @@
+  rnVersion: string,
+) {
+  const validVersion = semver.valid(rnVersion);
+  if (validVersion) {
+    const major = semver.major(validVersion);
+    const minor = semver.minor(validVersion);
+    if (major === 0 && minor >= 60) {
+      return `^${major}.${minor}.0-0`;
+    }
+  }
+  ErrorOutOnUnsupportedVersionOfReactNative(rnVersion);
+}
+
+function getMatchingReactNativeSemVerForReactNativeWindowsVersion(
+  rnwVersion: string,
+) {
+  const validVersion = semver.valid(rnwVersion);
+  if (validVersion) {
+    const major = semver.major(validVersion);
+    const minor = semver.minor(validVersion);
+    if (major === 0 && minor >= 59) {
+      return `^${major}.${minor}`;
+    }
+  }
+  return 'unknown';
+}
+
+function getLatestMatchingVersion(
+  pkg: string,
+  versionSemVer: string,
+): Promise<string> {
+  return new Promise((resolve, reject) => {
+    if (semver.validRange(versionSemVer)) {
+      npm.packages.releases(
+        pkg,
+        (err: any, details: {[key: string]: object}) => {
+          if (err) {
+            reject(err);
+          } else if (details) {
+            const versions = Object.keys(details);
+            if (versions.length > 0) {
+              const candidates = versions
+                .filter(_ => semver.satisfies(_, versionSemVer))
+                .sort(semver.rcompare);
+              if (candidates && candidates.length > 0) {
+                resolve(candidates[0]);
+                return;
+              }
+            }
+          }
+          reject(
+            new Error(`No matching version of ${pkg}@${versionSemVer} found`),
+          );
+        },
+      );
+    } else {
+      // Assume that versionSemVer is actually a tag
+      npm.packages.release(
+        pkg,
+        versionSemVer,
+        (err: any, details: {version: string}[]) => {
+          if (err) {
+            reject(err);
+          } else if (details && details.length > 0) {
+            resolve(details[0].version);
+            return;
+          }
+          reject(
+            new Error(`No matching version of ${pkg}@${versionSemVer} found`),
+          );
+        },
+      );
+    }
+  });
+}
+
+async function getLatestMatchingRNWVersion(
+  versionSemVer: string,
+): Promise<string> {
+  try {
+    const version = await getLatestMatchingVersion(
+      'react-native-windows',
+      versionSemVer,
+    );
+    if (!version) {
+      console.error(
+        `Error: No version of react-native-windows@${versionSemVer} found`,
+      );
+      process.exit(EXITCODE_NO_MATCHING_RNW);
+    }
+    return version;
+  } catch (err) {
+    console.error(
+      `Error: No version of react-native-windows@${versionSemVer} found`,
+    );
+    process.exit(EXITCODE_NO_MATCHING_RNW);
+  }
+}
+
+/**
+ * Check if project is using Yarn (has `yarn.lock` in the tree)
+ */
+function isProjectUsingYarn(cwd: string) {
+  return findUp.sync('yarn.lock', {cwd});
+}
+
+async function doMain() {
```

Comment:
nit: it's pretty common to also just do:

```javascript
(async () => {
// Main body here
})();
```
If this was modeled after `override-tools`, I had this so I could wrap the world with a lock.